Pre commit

roberth opened the pull request:
Adds pre-commit, shell hooks, activation hooks, nixpkgs-fmt.

I will look into extracting the hook modules into nix-pre-commit-hooks; changing that repo to be compatible with project.nix.

TODO
 - [x] bump to master
 - [x] add TODOs mentioning #14
 - [x] remove `formatters.nixpkgs-fmt`

Closes #1

roberth committed:
Add shell.hooks

roberth committed:
Add pre-commit, first iteration

roberth committed:
Make config.root always a string for store-safety

roberth committed:
Add activation.hooks for pre-commit install

roberth committed:
pre-commit: Generate the config file too

roberth committed:
pre-commit: git diff: +color -paging

roberth committed:
pre-commit: Run in Nix

roberth committed:
Doc fixes

roberth committed:
Add formatters.nixpkgs-fmt

roberth committed:
Enable nixpkgs-fmt

roberth committed:
niv: Exclude sources.nix from formatting

roberth committed:
Format

roberth committed:
Move pre-commit out (mostly)

roberth committed:
sources.nix-pre-commit-hooks: Update to master

roberth committed:
Remove formatters.nixpkgs-fmt options

roberth committed:
Don't run activation in lorri sandboxed shellHook

roberth committed:
niv: Use pre-commit global excludes

domenkozar commented on modules/pre-commit.nix:
```diff
@@ -0,0 +1,49 @@
+      '';
+    };
+
+    enableAutoInstall = mkOption {
```
Why would someone set this to false?

roberth replied:
In case it doesn't work for them for whatever reason. It was supposed to be an escape hatch, but I think `activation.enableShellHook` is already good enough as an escape hatch. I think I wrote that one after this one. Will remove this one :+1: .

roberth committed:
Remove pre-commit.enableAutoInstall escape hatch

Any problems with auto-install should be addressed in the
activation.* options.
If it turns out someone needs these, we can reintroduce it.
For now, we keep it simple.

roberth committed:
sources.nix-pre-commit-hooks: global-exclude -> master
